lacework/resource_lacework_integration_ghcr.go

```diff
@@ -0,0 +1,265 @@
+				Computed: true,
+			},
+			"created_or_updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"org_level": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkIntegrationGhcrCreate(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	data := api.NewContainerRegistry(d.Get("name").(string),
+		api.GhcrContainerRegistry,
+		api.GhcrData{
+			LimitByTag:            castAttributeToStringSlice(d, "limit_by_tags"),
+			LimitByLabel:          castAttributeToArrayOfKeyValueMap(d, "limit_by_label"),
+			LimitByRep:            castAttributeToStringSlice(d, "limit_by_repositories"),
+			LimitNumImg:           d.Get("limit_num_imgs").(int),
+			RegistryNotifications: d.Get("registry_notifications").(bool),
+			Credentials: api.GhcrCredentials{
+				Username: d.Get("username").(string),
+				Password: d.Get("password").(string),
+				Ssl:      d.Get("ssl").(bool),
+			},
+		},
+	)
+
+	if !d.Get("enabled").(bool) {
+		data.Enabled = 0
+	}
+
+	log.Printf("[INFO] Creating ContVulnCfg integration for %s registry type with data:\n%+v\n",
+		api.GhcrContainerRegistry.String(), data)
+	response, err := lacework.V2.ContainerRegistries.Create(data)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.IntgGuid)
+	d.Set("name", response.Data.Name)
+	d.Set("intg_guid", response.Data.IntgGuid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.CreatedOrUpdatedBy)
+	d.Set("type_name", response.Data.Type)
+	d.Set("org_level", response.Data.IsOrg == 1)
+
+	log.Printf("[INFO] Created ContVulnCfg integration for %s registry type with guid %s\n",
+		api.GhcrContainerRegistry.String(), response.Data.IntgGuid)
+	return nil
+}
+
+func resourceLaceworkIntegrationGhcrRead(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Reading ContVulnCfg integration for %s registry type with guid %s\n",
+		api.GhcrContainerRegistry.String(), d.Id())
+	response, err := lacework.V2.ContainerRegistries.GetGhcr(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", response.Data.Name)
+	d.Set("intg_guid", response.Data.IntgGuid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.CreatedOrUpdatedBy)
+	d.Set("type_name", response.Data.Type)
+	d.Set("org_level", response.Data.IsOrg == 1)
+
+	creds := struct {
+		username string
+		password string
+		ssl      bool
+	}{username: response.Data.Data.Credentials.Username,
+		password: response.Data.Data.Credentials.Password,
+		ssl:      response.Data.Data.Credentials.Ssl}
+	d.Set("credentials", creds)
+	d.Set("registry_domain", response.Data.Data.RegistryDomain)
```

Comment:
I don't see this defined in the Schema.... 🤔 This should be a computed attribute.

Reply:
this is not defined... 😂